source/common/protobuf/utility.cc

```diff
@@ -424,6 +480,22 @@ std::string MessageUtil::CodeEnumToString(ProtobufUtil::error::Code code) {
   }
 }
 
+ProtobufWkt::Value ValueUtil::loadFromYaml(const std::string& yaml) {
+  try {
+    return parseYamlNode(YAML::Load(yaml));
```

Comment:
Can we limit the recursion depth in YAML parsing?

Reply:
I don't see any easy way with yaml-cpp library.

Reply:
Are you thinking about the "billion laughs" attack that created a high severity CVE for Kubernetes @htuch ? https://github.com/kubernetes/kubernetes/issues/83253

Reply:
Something like that. I think if we can move to enforcing YAML to be local CLI or filesystem only, I'm more comfortable with there not being a bound here.

Reply:
@lizan as a follow up can you de-dup this non-trivial exception handling code with the same code above? It should be a shared utility function if possible. Thank you.

Reply:
@mattklein123 https://github.com/envoyproxy/envoy/pull/9021#discussion_r346663622 it is not that easy because the `try` part is different.

Reply:
Can't the `YAML::Load` part just be wrapped with a function that deals with the exceptions?

Reply:
Not only `YAML::Load`, the `parseYamlNode` also throws almost same set of exceptions. It's not easy to refactor unless we do something javascripty (taking callback and try), which I guess Harvey will hate.

Reply:
OK I see. I would be fine with passing a lambda in and calling it, but it's not a big deal.